Tests/UnitTests/Core/Seeding/HoughTransformTest.cpp

```diff
@@ -0,0 +1,135 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Seeding/HoughTransformUtils.hpp"
+#include "ActsExamples/EventData/DriftCircle.hpp"
+#include "ActsExamples/EventData/MuonSimHit.hpp"
+
+#include <array>
+#include <memory>
+#include <vector>
+
+namespace Acts::Test {
+
+BOOST_AUTO_TEST_CASE(hough_transform_seeder) {
+  // we are using the slope on yz plane with the y coordinate (hardcoded from
+  // the csv MuonSimHit data)
+  std::vector<std::pair<double, double>> simHits = {
+      {-0.0401472 / 0.994974, -422.612}};
+  // The drift circles info (hardcoded from the csv file)
+  std::vector<std::vector<float>> tubePos{{-4.54747e-13, -427.981, -225.541},
+                                          {0, -412.964, -199.53},
+                                          {1.36424e-12, -427.981, -173.519},
+                                          {-9.09495e-13, -427.981, 173.519},
+                                          {0, -442.999, 199.53},
+                                          {-4.54747e-13, -427.981, 225.541}};
+  std::vector<float> driftRadius{14.5202, 1.66237, 12.3176,
+                                 1.5412,  12.3937, 3.77967};
+  std::vector<std::vector<std::int8_t>> stationInfo{
+      {5, -1, 4, 1, 1, 39}, {5, -1, 4, 1, 2, 38}, {5, -1, 4, 1, 3, 39},
+      {5, -1, 4, 2, 1, 39}, {5, -1, 4, 2, 2, 39}, {5, -1, 4, 2, 3, 39}};
+
+  BOOST_CHECK_EQUAL(tubePos.size(), driftRadius.size());
+  BOOST_CHECK_EQUAL(stationInfo.size(), tubePos.size());
+  // Define the drift Circles
+  ActsExamples::DriftCircleContainer driftCircles;
+  for (std::size_t i = 0; i < tubePos.size(); i++) {
+    ActsFatras::Hit::Vector3 tube_pos{tubePos[i][0] * Acts::UnitConstants::mm,
+                                      tubePos[i][1] * Acts::UnitConstants::mm,
+                                      tubePos[i][2] * Acts::UnitConstants::mm};
+    driftCircles.push_back(ActsExamples::DriftCircle(
+        std::move(tube_pos), driftRadius[i], 0.0f, stationInfo[i][0],
+        stationInfo[i][1], stationInfo[i][2], stationInfo[i][3],
+        stationInfo[i][4], stationInfo[i][5]));
+  }
+
+  // configure the binning of the hough plane
+  Acts::HoughTransformUtils::HoughPlaneConfig planeCfg;
+  planeCfg.nBinsX = 1000;
```

Comment:
could make this smaller for the CI in case memory is a an issue (maybe not a problem)